### PR TITLE
Template for unit tests

### PR DESCRIPTION
**Pull request by RondeauG**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - Not really (at least not enough to resolve the issue)
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] This PR does not seem to break the templates.
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Implements a working template on which to start creating unit tests for xscen.
* Added NotImplementedError in `climatological_mean` and `compute_deltas` when using daily data.

### Does this PR introduce a breaking change?

* No.

### Other information:

This is a start for #9

### CODE REVIEW

**Comment by RondeauG** [comment]
@aulemahal @Zeitsperre Feel free to work directly on this branch and to change/remove what I've done. Once we're happy with the formatting, we'll merge it and I'll start making real unit tests.

File: tests/conftest.py
Comment (Zeitsperre) [inline tests/conftest.py:50]: I think we should explore the xclim-based solution. Would you want me and/or @aulemahal to add something there first?

Comment (RondeauG) [reply tests/conftest.py:50]: Yeah, this is just what I had at the time of our discussion, and helps to show the modifications I had to do (such as expose `freq`). We should indeed work on modifying `xclim` first.

Comment (RondeauG) [reply tests/conftest.py]: @Zeitsperre @aulemahal 
This new commit will fail because `test_timeseries` isn't yet accessible outside of `xclim`'s master branch, but is this closer to what we had in mind ? Or did we want to remove fixtures altogether ?

File: tests/conftest.py
Comment (aulemahal) [inline tests/conftest.py:33]: This is the xclim solution, where we avoided a huge PR by keeping the same fixture. However, I'm not sure I see any gain in using fixtures here ?

I would have suggested importing `test_timeseries` in each test file and using the function directly in the tests. This will  make for a simpler `conftest.py`.

Comment (RondeauG) [reply tests/conftest.py:33]:
```
import numpy as np
import xscen as xs
from xclim.testing.helpers import test_timeseries
import pytest


class TestClimatologicalMean:
    def test_daily(self):
        ds = test_timeseries(np.tile(np.arange(1, 13), 3), variable="tas", start="2001-01-01", freq="D", as_dataset=True)
        with pytest.raises(NotImplementedError):
            xs.climatological_mean(ds)
```
I get this error when trying to use it directly:
```
___________________________________________________________________________________________________________________ ERROR at setup of test_timeseries ____________________________________________________________________________________________________________________
  def test_timeseries(
E       fixture 'values' not found
>       available fixtures: anyio_backend, anyio_backend_name, anyio_backend_options, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cleanup_notebook_data_folder, cov, doctest_namespace, include_metadata_in_junit_xml, json_metadata, metadata, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, tas_series, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, xdoctest_namespace
>       use 'pytest --fixtures [testpath]' for help on them.

```

Comment (aulemahal) [reply tests/conftest.py:33]: Je'l savais merde.

```
from xclim.testing.helpers import test_timeseries as timeseries
```

When a member function of a `test.py` file starts with `test_` it is considered a test and pytest tries to run it so. `values` is the first arg of the function and pytest assumes it's a fixture.

**Comment by RondeauG** [comment]
What I'm uploading here is a version that uses `test_timeseries` and hits pretty much all of `climatological_mean`. Since this is the best time to establish a format, I'm open to all suggestions!

File: tests/test_aggregate.py
Comment (aulemahal) [inline tests/test_aggregate.py]: what ? Standard -> Datetime360Day ?

My commit changed the logic here anyway.

Comment (RondeauG) [reply tests/test_aggregate.py]: I think that's some bad copy-paste 😄

**Comment by aulemahal** [comment]
@RondeauG 
Please see the changes introduced in my latest commit. 

- Instead of looping inside the test function, I used `pytest.mark.parametrize`. The benefit here is that if a single element fails:
    - The others are also run
    - The pytest error output specifies the exact element causing the bug
    - And we can run a single element by hand if needed, while debugging.
- I switched to conventionnal `assert` when we are comparing scalars.
- I rewrote a check with a `set` logic instead of an `all` of a in comprehension list.
- Rewrote the calendar check to use only xarray properties and pytest parametrize

**Comment by RondeauG** [comment]
This should be good for review.

**Review by Zeitsperre — APPROVED** [review]
Will wait on @aulemahal's approval as well

File: xscen/aggregate.py
Comment (aulemahal) [inline xscen/aggregate.py:190]: This doesn't fit the docstring, which says None is possible!
I'm guessing you want to change the docstring?

**Review by aulemahal — APPROVED** [review]
One comment to the changes in `climatological_mean`.

For the test, I think this is a good example of what we want. Even, it is on the exhaustive side, which is good.

Warning : all tests here are done on artificial data, and thus it may fail to tests some real-life caveats. When adding more tests, we could make use of xclim's test data too.

File: xscen/aggregate.py
Comment (RondeauG) [reply xscen/aggregate.py:190]: Ah, true! I got confused because the default is a string, not `None`. I'll reverse that.

**Pull request merged**
